bolt-design-system/bolt
# Fix ratio calculation bug

## Jira

 - http://vjira2:8080/browse/BDS-977
 - http://vjira2:8080/browse/BDS-1154

## Summary

 - Fixed the ratio bug that sometimes miscalculated aspect ratio
 - Added tests for logo, where we first noticed the bug

## Details
Previously, in `ratio.twig` we were dividing ratio height or width values by 100 when the value was > 100. This worked fine when both values were either > 100 or < 100 but failed when one was above and the other below. The error would miscalculate the aspect ratio, causing the visual regression we saw with the logo.

Instead of dividing by 100, I'm now getting the greatest common divisor of our ratio values (via a new twig function). I added VR tests for logo and updated the tests to use the new reduced fractions.

## How to test

- Run tests locally to ensure they pass.
- Review the following files:
  - [core-php/src/TwigFunctions.php](https://github.com/bolt-design-system/bolt/pull/1123/files#diff-ba05946309e999248b59043efb61b3bf)
  - [bolt-ratio/src/ratio.twig](https://github.com/bolt-design-system/bolt/pull/1123/files#diff-b7457575f6ff994e1ef90e187f49d180)
  - [bolt-logo/\_\_tests\_\_/logo.js](https://github.com/bolt-design-system/bolt/pull/1123/files#diff-f53843c07595b9cecf31bcb38507875e)
  - [bolt-logo/TESTING.md](https://github.com/bolt-design-system/bolt/pull/1123/files#diff-2d600f66a876a3553622e0d06870d738)
